server/src/main/java/com/ibm/ws/lars/rest/PersistenceBean.java

```diff
@@ -262,27 +269,24 @@ private BasicDBObject createFilterObject(String field, Condition condition) {
             logger.fine("query: pagination object " + pagination);
         }
 
-        List<DBObject> results = new ArrayList<DBObject>();
-        try (DBCursor cursor = getAssetCollection().find(filterObject, projectionObject)) {
-            if (logger.isLoggable(Level.FINE)) {
-                logger.fine("query: found " + cursor.count() + " assets.");
-            }
-
-            if (pagination != null) {
-                cursor.skip(pagination.getOffset());
-                cursor.limit(pagination.getLimit());
-            }
+        DBCursor cursor = getAssetCollection().find(filterObject, projectionObject);
+        if (logger.isLoggable(Level.FINE)) {
+            logger.fine("query: found " + cursor.count() + " assets.");
+        }
 
-            if (sortObject != null) {
-                cursor.sort(sortObject);
-            }
+        if (pagination != null) {
+            cursor.skip(pagination.getOffset());
+            cursor.limit(pagination.getLimit());
+        }
 
-            for (DBObject obj : cursor) {
-                convertObjectIdToHexString(obj);
-                results.add(obj);
-            }
+        if (sortObject != null) {
+            cursor.sort(sortObject);
         }
-        return results;
+
+        AssetCursor result = new MongoAssetCursor(cursor);
+        result.addOperation(CONVERT_OID_TO_HEX);
```

Comment:
Hmm, possibly? Or I could put this functionality into `MongoAssetCursor` itself on the basis that we will always need to convert the id when turning a MongoDB record into an asset.

On the other hand, it fits quite nicely into an operation.